pkg/cluster/fetch.go

```diff
@@ -22,6 +22,9 @@ type EksdReleaseFetch func(ctx context.Context, name, namespace string) (*eksdv1
 
 type OIDCFetch func(ctx context.Context, name, namespace string) (*v1alpha1.OIDCConfig, error)
 
+// BuildSpec constructs a cluster.Spec for a eks-a cluster by retrieving all
+// necessary objects using fetch methods
+// This is deprecated in favour of BuildSpec
 func BuildSpecForCluster(ctx context.Context, cluster *v1alpha1.Cluster, bundlesFetch BundlesFetch, eksdReleaseFetch EksdReleaseFetch, gitOpsFetch GitOpsFetch, fluxConfigFetch FluxConfigFetch, oidcFetch OIDCFetch) (*Spec, error) {
```

Comment:
We should get rid of this method now that we have the builder that uses a client. But I would leave that refactor for later to not blow up this PR